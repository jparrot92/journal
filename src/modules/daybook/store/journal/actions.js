import journalApi from '@/api/journalApi'
// export const myAction = async ({ commit }) => {

// }

export const loadEntries = async ({ commit }) => {

    const { data } = await journalApi.get('/entries.json')

    if ( !data ){
        commit('setEntries', [] )
        return
    }

    const entries = []
    for( let id of Object.keys( data ) ) {
        entries.push({
            id,
            ...data[id]
        })
    }

    commit('setEntries', entries )
}

export const updateEntry = async (/*{ commit }*/) => {

}


export const createEntry = async (/*{ commit }*/) => {

}
