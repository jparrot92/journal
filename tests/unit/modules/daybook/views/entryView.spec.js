import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'

import journal from '@/modules/daybook/store/journal'
import EntryView from '@/modules/daybook/views/EntryView.vue'

import { journalState } from '../../../mock-data/test-journal-state'

import Swal from 'sweetalert2'

const createVuexStore = (initialState) => createStore({
    modules: {
        journal: {
            ...journal,
            state: { ...initialState }
        }
    }
})

jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
}))

describe('Pruebas en el EntryView', () => {

    const store = createVuexStore( journalState )
    store.dispatch = jest.fn()

    const mockRouter = {
        push: jest.fn()
    }
    
    let wrapper

    beforeEach(() => {
        // Limpia los mock caza vez que ejecuta una prueba
        jest.clearAllMocks()
        wrapper = shallowMount( EntryView, {
            props: {
                id: '-NDF1PyxyfLxOtMc_1gW',
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            }
        })
    })
    
    test('debe de sacar al usuario porque el id no existe', () => {

        shallowMount( EntryView, {
            props: {
                id: 'Este ID no existe en el STORE',
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            }
        })

        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })

    })


    test('debe de mostrar la entrada correctamente', () => {

        expect( wrapper.html() ).toMatchSnapshot()
        expect(mockRouter.push).not.toHaveBeenCalled()

    })

    test('debe de borrar la entrada y salir', () => {

        Swal.fire.mockReturnValueOnce( Promise.resolve({ isConfirmed: true }) )

        wrapper.find('.btn-danger').trigger('click')

        expect(Swal.fire).toHaveBeenCalledWith({
            title: '¿Está seguro?',
            text: 'Una vez borrado, no se puede recuperar',
            showDenyButton: true,
            confirmButtonText: 'Si, estoy seguro'
        })
            
        setTimeout( () => {
            
            expect( store.dispatch ).toHaveBeenCalledWith('journal/deleteEntry', '-MfKM6PrX3s9QqURdLx5')
            expect( mockRouter.push ).toHaveBeenCalled()

        }, 1 )

    })

})