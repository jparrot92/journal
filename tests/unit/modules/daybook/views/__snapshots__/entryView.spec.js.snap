// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">30</span><span class="mx-1 fs-3">Septiembre</span><span class="mx-2 fs-4 fw-light">2022, Viernes</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedió hoy?"></textarea></div>
<img src="https://res.cloudinary.com/dv5fhqjdo/image/upload/v1664579042/vpubxhkfovxbd40dpghm.webp" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
